src/web/utils/apiError.tsx

```diff
@@ -25,3 +27,18 @@ export function backendError(e: unknown, overrideMessage: string) {
   }
   return Error(overrideMessage);
 }
+
+export function handleErrorPopup(
+  e: Error,
+  setStatusPopup: React.Dispatch<React.SetStateAction<StatusNotificationType | undefined>>,
+  setShowStatusPopup: React.Dispatch<React.SetStateAction<boolean>>
+) {
+  const hasHash = Object.hasOwn(e, 'errorHash') && (e as ApiError).errorHash;
+  const hash = hasHash ? `: (${(e as ApiError).errorHash})` : '';
+  setStatusPopup({
+    type: 'Error',
+    message: `${e.message}${hash}`,
+  });
+  setShowStatusPopup(true);
+  throw new Error(e.message);
```

Comment:
This loses the call stack of the original exception and re-creates it here - if it's a real exception, you should either re-throw the original exception (to preserve the stack trace) or provide the original exception to the Error constructor - see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#rethrowing_an_error_with_a_cause

Reply:
good point - I'll just throw the original error: `throw e`.